fsi/body.go

```diff
@@ -81,3 +82,74 @@ func GetBody(dirPath string, format dataset.DataFormat, fcfg dataset.FormatConfi
 
 	return dsio.ConvertFile(file, structure, st, limit, offset, all)
 }
+
+// GetBodyAsInterface is an FSI version of base.ReadBodyAsInterface
+func GetBodyAsInterface(dirPath string, offset, limit int, all bool) (interface{}, error) {
+	components, err := getAllComponents(dirPath)
+	bodyComponent := components.Base().GetSubcomponent("body")
+	f, err := os.Open(bodyComponent.Base().SourceFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	structure, err := prepareInStructure(components, f)
+	if err != nil {
+		return nil, err
+	}
+
+	rr, err := dsio.NewEntryReader(structure, f)
+	if err != nil {
+		return nil, fmt.Errorf("error allocating reader: %s", err)
+	}
+	return base.ReadEntries(rr)
+}
+
+func getAllComponents(dirPath string) (component.Component, error) {
```

Comment:
This is totally fine, just wanted to point out that this is likely a result of `component.Component` having an unwieldy API. These two calls should probably be combined into one, which would remove the need for this `getAllComponents` function. Personal TODO for myself.